code/game/turfs/simulated/chasm.dm

```diff
@@ -11,6 +11,7 @@
 	canSmoothWith = list(/turf/open/floor/fakepit, /turf/open/chasm)
 	density = TRUE //This will prevent hostile mobs from pathing into chasms, while the canpass override will still let it function like an open turf
 	var/static/list/falling_atoms = list() //Atoms currently falling into the chasm
+	var/static/list/forbidden_types = typecacheof(list(/obj/effect/portal, /obj/singularity, /obj/structure/stone_tile, /obj/item/projectile))
```

Comment:
Should be all of obj/effect tbh instead of just portal

Reply:
foam should fall down the chasm

and why would smoke fall down it

Reply:
that'd cause like, blood to not fall.